### PR TITLE
Add `_darwin.go' files so u-root will work on MacOS

### LINKED ISSUE #1254: ldd: LdSo() undefined when building on OSX
# Summary
When `go get`ing or `go build`ing the repository on OSX, the following error occurs:

```go
# github.com/u-root/u-root/pkg/ldd
pkg/ldd/ldd_unix.go:158:18: undefined: LdSo
```

## Specs
**Go version:** go1.12.5 darwin/amd64
**OS version:** macOS Mojave 10.14.5 (18F132)

### PR DESCRIPTION
**Pull request by synackd**
Fixes #1254 

@rminnich We talked about file renaming to enable u-root to build and run on MacOS. This patch does just that.

### CODE REVIEW

File: cmds/core/lsmod/lsmod_darwin.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
```
Comment (rminnich) [inline cmds/core/lsmod/lsmod_darwin.go]: This can't be right. There is no /proc on darwin.

File: cmds/core/kexec/kexec_darwin.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
+// Copyright 2015-2018 the u-root Authors. All rights reserved
```
Comment (rminnich) [inline cmds/core/kexec/kexec_darwin.go]: darwin supports kexec? That's news to me.

File: cmds/core/installcommand/installcommand_darwin.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
+// Copyright 2017 the u-root Authors. All rights reserved
```
Comment (rminnich) [inline cmds/core/installcommand/installcommand_darwin.go]: so does installcommand work with a darwin kernel?

File: cmds/core/insmod/insmod_darwin.go [hunk 4]
```diff
@@ -0,0 +1,42 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
```
Comment (rminnich) [inline cmds/core/insmod/insmod_darwin.go]: This file is called _darwin but has a comment about Linux? This can't be right.

**Comment by synackd** [comment]
Hmm... It seems that u-root will now build and generate an initramfs even without `lsmod_darwin.go`, `insmod_darwin.go`, and `kexec_darwin.go`, which I simply copied from `*_linux.go`. Perhaps this is due to a recent rebase onto master.

**Comment by synackd** [comment]
...and `rmmod_darwin.go`.

**Comment by synackd** [comment]
@rminnich By "author tags" are you referring to the comment block at the top of the file e.g:
```go
// Author:
//     Roland Kammerer <dev.rck@gmail.com>
```

File: cmds/core/insmod/insmod_darwin.go [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// Insert a module into the Linux kernel
+//
+// Synopsis:
```
Comment (rminnich) [inline cmds/core/insmod/insmod_darwin.go]: The bigger issue is that you have created files that claim to implement operations on darwin that do no exist. For example, there is no switch_root on darwin. That does not exist. Plus, you have files named _darwin.go that reference Linux. This just makes no sense. I appreciate the work but it's not solving the problem the right way.

**Comment by synackd** [comment]
@rminnich Ah, I see the problem. I initially understood this as trying to build u-root *for* Linux *on* Darwin, but now I understand that this is indeed building u-root *for* Darwin. My mistake. That is why you see all of the Linux stuff in these files.

**Comment by hugelgupf** [comment]
No worries. Go has native cross-compilation, so if you wanted to build u-root for Linux on Darwin: `GOOS=linux ./u-root ...` (same as with the go tool, `GOOS=linux go build ...`)

**Comment by synackd** [comment]
That said, I see it fit to close this pull request and someone can open a new one and solve this properly. Should I close #1254 and open a new one with an appropriate subject (i.e. "Add Darwin support")?

**Comment by synackd** [comment]
Thank you, @hugelgupf! Sorry for the confusion...

**Pull request closed without merging**